nuxt/framework
# v3.0.0-rc.5

#### ⚠️  Breaking Changes

This release contains slight breaking changes. Please check the linked pull-requests for notes.

- **auto-import:** ⚠️Unwrap auto imports in SFC templates (#5573)
- **nuxt:** ⚠️  Support universal global middleware (#5038)
- **nuxt:** ⚠️  Remove legacy app context (#5630)


### 🚀 Enhancements

  - **kit:** Allow `addServerHandler` to use method suffix of the file path (#5465)
  - **kit:** Support client and server flags for `addVitePlugin` (#5560)
  - **auto-import:** ⚠️ Unwrap auto imports in SFC templates (#5573)
  - **kit:** Add `addLayout` utility (#5537)
  - **nuxt:** ⚠️ Support global middleware without router (#5038)
  - **nuxt:** Allow passing custom fetch options to `useFetch` (#5660)
  - **nuxt:** Prerender all pages by default (#5709)
  - **nuxt:** Tree-shake client and server-only composables (#5749)
  - **nuxt:** Automatically generate unique keys for keyed composables (#4955)
  - **nuxt:** Support `custom` prop for `<nuxt-link>` (#4249)
  - **nuxt:** Add `<NuxtLoadingIndicator>` component [WIP] (#5121)


### 🔥 Performance

  - **nuxt:** ⚠️  Remove legacy app context (#5630)


### 🩹 Fixes

  - **schema:** Add types for `vite:` hooks (#5472)
  - **nuxt:** Normalize string transpile paths for windows (#5461)
  - **nuxt:** Prefer unctx context over `getCurrentInstance` for `useNuxtApp` (#5456)
  - **vite:** Add typecheck to vite plugins (#5464)
  - **vite:** Disable HMR in build (#5505)
  - **schema, postcss:** `cssnano` is enabled if not dev (#5583)
  - **vite:** Do not watch ignored pathes (#5632)
  - **vite-node:** On-demand manifest (#5017)
  - **nuxt:** Pass attrs and props to client-only components (#5668)
  - **kit:** Allow adding server-only and client-only components separately (#5686)
  - **webpack:** Transpile `vue-demi` (#5721)
  - **nuxt:** `useRequestHeaders` type should allow for `undefined` values (#5748)
  - **nuxt:** Update `useFetch` key warning to include any function or blob (#5710)
  - **vite:** Do not inline dynamic imports in server (#5506)
  - **nuxt:** Handle `undefined` when applying defaults for `useAsyncData` (8d2f8057)
  - **vite:** Fix `vite-node` circular reference (#5764)
  - **nuxt:** Use virtual server stub for `ssr: false` (#5773)
  - **nuxt:** Don't return stub function from tree-shaken composables (#5786)
  - **test-utils:** Update vitest import path with `.mjs` (#5817)
  - **nuxt:** Only tree-shake composables on their own lines (#5850)
  - **nuxt:** Extend nitro routes instead of overriding (#5828)
  - **cli:** Show base url in terminal (#5337)
  - **nuxt:** Add import protection to nitro config (#5847)
  - **nuxt:** Replace `process.dev` in nitro bundle (#5852)
  - **nuxt:** Warn about legacy and invalid plugins (#5857)
  - **nuxt:** Fix issues with `vue-router` and `@vue/devtools-api` bundling (#5874)
  - **vite:** Close vite watcher before building (#5875)

### 📖 Documentation

  - Update roadmap (#5443)
  - Rewrite note wording (#5462)
  - Align value for defaulting the page's title with type declaration (#5447)
  - Fix link to `nuxt/image` issue (#5490)
  - Add notes about reporting security issues (#4895)
  - **testing:** Fix async/await (#5539)
  - **api:** Add watch parameter in `useFetch` options (#5562)
  - **examples:** Rename `hello.ts` to `[...hello].ts` (#5261)
  - Update usage about runtime config and environment variables (#5569)
  - Update `pnpm` usage (#5597)
  - Minor wording improvements (#5622)
  - Replace invalid link to composables (#5611)
  - Rework the layout section (#5118)
  - **api:** Add docs for `<NuxtPage>` component (#5591)
  - Clarify that `definePageMeta` only works in the pages directory (#5663)
  - Update vuejs-development (#5679)
  - Add full stop (#5680)
  - **api:** Update commands with new `NODE_ENV` behavior (#5739)
  - **plugins:** Add directive example (#5667)
  - Fix Vite check for Nuxt 3 (#5768)
  - `useFetch` uses isomorphic `$fetch` (#5769)
  - Add content directory to directory-structure (#5755)
  - **api:** Add `anavigateTo` docs (#5701)
  - Fix link to `NuxtLoadingIndicator`  component src (#5776)
  - Use `defineLazyHandler` in wasm example (#5797)
  - Use `toUTCString` in `useAsyncData` example (#5798)
  - Add missing dot (#5815)
  - Update `nuxt-link` (#5819)
  - Lint whitespaces (#5633)
  - Fix typo (#5865)
  - Fix typo (#5863)

### ❤️  Contributors

- Ahad Birang
- Alexander Lichter
- AnPineau
- Anders Søgaard
- Anthony Fu
- Bob Olde Hampsink
- Bot08
- Clément Ollivier
- Cupid Valentine
- Damian
- Daniel Roe
- Donatelloraphael
- Francisco Buceta
- Haruaki OTAKE
- Ibrahim Abdullahi Aliyu
- Julien Huang
- Krutie Patel
- Mourad EL CADI
- Mrauhu
- Nestor Vera
- Paranoid
- Pooya Parsa
- Qin Guan
- Sumiren
- Sébastien Chopin
- Thibault Vlacich
- Thomas Jowsey
- Vasiliy
- Xanlantos
- Yuki Inoue
- 井上裕貴
----

Changes from **v3.0.0-rc.4...chore/rc.5**

See all changes: https://github.com/nuxt/framework/compare/v3.0.0-rc.4...chore/rc.5
